app/controllers/reservations_controller.rb

```diff
@@ -183,6 +183,15 @@ def update
       return
     end
 
+    if !current_user.administrator?
```

Comment:
It seems like this should come after the `assign_times_from_params` call. I could be wrong, but without that, it probably won't maintain your changes you made through the interface.